notifier/notifier.go

```diff
@@ -144,15 +151,32 @@ func (n *Notifier) Start(closeChan chan struct{}) {
 	// Check Redis for jobs left in InProgress state
 	n.collectRogueCallbacks()
 
-	ctx, cancelfunc := context.WithCancel(context.Background())
 	go n.stats.Run(ctx)
 
+	// Start monitoring delivery reports for each backend
+	var deliveriesWg sync.WaitGroup
+	for id := range n.backends {
+		deliveriesWg.Add(1)
+		go func(id string) {
+			n.MonitorDeliveries(ctx, id)
+			deliveriesWg.Done()
```

Comment:
Better `defer` this to ensure it is always called.

Reply:
sure!